includes/html/api_functions.inc.php

```diff
@@ -1084,6 +1084,45 @@ function get_port_info(Illuminate\Http\Request $request)
     });
 }
 
+function update_port_description(Illuminate\Http\Request $request)
+{
+    $port_id = $request->route('portid');
+    $port = Port::hasAccess(Auth::user())
+        ->where([
+            'port_id' => $port_id,
+        ])->first();
+    if (empty($port)) {
+        return api_error(400, 'Invalid port ID.');
+    }
+
+    $data = json_decode($request->getContent(), true);
+    $field = 'description';
+    $content = $data[$field];
+
+    if (empty($content)) {
+        return api_error(400, 'New port description has not been supplied.');
+    }
+
+    $port->ifAlias = $content;
```

Comment:
This won't work as expected, it will be overwritten by the poller.  You would need to set a device attribute to ensure it is not overwritten.

Reply:
OK, I'll look into this. I wasn't sure if it would be overwritten by the poller, as you can set the description for a port through the web UI in Port Settings, but if this would be overwritten then I'll look into a better solution for making sure that doesn't happen.

Reply:
Look at the code for overriding it in the webui, should show you how to do it.

Reply:
> Look at the code for overriding it in the webui, should show you how to do it.

so, just to make sure I'm understanding this right - I'm looking at `/includes/html/forms/update-ifalias.inc.php` - I set the `ifAlias` as I have been doing, but then I also set a flag in `devices_attribs`, and that flag tells the poller to not overwrite the port description?

I can also see that the webUI logs when an interface description is updated - I'll make sure to add that too :)

```php
set_dev_attrib($device, 'ifName:' . $ifName, 1);
```
```
+-----------+-----------+-------------------+--------------+---------------------+
| attrib_id | device_id | attrib_type       | attrib_value | updated             |
+-----------+-----------+-------------------+--------------+---------------------+
|       286 |       154 | ifName:lag-4/fp-3 | 1            | 2023-11-13 16:30:59 |
+-----------+-----------+-------------------+--------------+---------------------+
```

Reply:
I've changed `update_port_description` to mimic the behaviour of the web UI, so it shouldn't be overwritten by the poller now. If I've made any mistakes anywhere, let me know.

Reply:
@murrant Hi, would it be possible to request a review on this? I requested a review last Monday but haven't heard back from you and GitHub won't allow me to request again.